providers/base/bin/snap_update_test.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# All rights reserved.
+#
+# Written by:
+#    Patrick Liu <patrick.liu@canonical.com>
+#    Pierre Equoy <pierre.equoy@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+from glob import glob
+import json
+import logging
+import os.path
+import sys
+import time
+
+from checkbox_support.snap_utils.snapd import Snapd
+
+
+def guess_snaps() -> dict:
+    """
+    Guess the names of the kernel, snapd and gadget snaps from installed snaps
+    on the system.
+
+    :return: a dict with the snap names for each snap found
+    :rtype: dict
+    """
+    snapd = Snapd()
+    installed_snaps = snapd.list()
+    snaps = {}
+    for snap in installed_snaps:
+        if snap["type"] == "kernel":
+            snaps["kernel"] = snap["name"]
+        elif snap["type"] == "gadget":
+            snaps["gadget"] = snap["name"]
+        elif snap["type"] == "snapd":
+            snaps["snapd"] = snap["name"]
+    return snaps
+
+
+def get_snap_base_rev() -> dict:
+    """
+    Retrieve the name and the base revision of each snap originally installed
+    on the system.
+
+    :return: a dict containing the snap names and their base revisions
+    :rtype: dict
+    """
+    base_snaps = glob("/var/lib/snapd/seed/snaps/*.snap")
+    base_rev_info = {}
+    for snap_path in base_snaps:
+        snap_basename = os.path.basename(snap_path)
+        snap_name = os.path.splitext(snap_basename)[0]
+        snap, rev = snap_name.rsplit("_", maxsplit=1)
+        base_rev_info[snap] = rev
```

Comment:
In `/var/lib/snapd/seed/snaps/`, there can only be one revision for a given snap, because it's the one that comes pre-installed when you install the image. Therefore, we will never run into this problem.

And I think if we had to deal with it, we could simply assume the base revision is the smallest int between the two.